mne/io/brainvision/tests/test_brainvision.py

```diff
@@ -102,6 +102,58 @@ def test_vmrk_meas_date():
     assert raw.info['meas_date'] is None
     assert 'unspecified' in repr(raw.info)
 
+    # Reuse our vmrk to test several other cases with dates
+    tmpdir = _TempDir()
```

Comment:
The modern way to do this is with a pytest `tmpdir` fixture like:
```
def test_whatever(tmpdir):
    temp_dir = str(tmpdir)  # because tmpdir does not get treated as str correctly by all functions, e.g. op.join
    ...
```

Reply:
wow, that's super helpful, thanks! I'll put the link here for my future self: http://doc.pytest.org/en/latest/tmpdir.html#the-tmpdir-fixture

Reply:
I would change it and find a way that all parts of the test look alike. or even parametrize the test.

But we can do that in a subsequent PR. @sappelhoff we can write it together or you can start it and I can push on it.

Reply:
yes, I noticed that when writing the tests, that this could benefit from a refactoring 👍 I can give it a try

Reply:
go for it I'll be happy to help.

Reply:
here's a search in places that could be similar. fixtures that create temporary files of some sort and are consumed by the tests:

```sh
~/code/mne-python kernel_param_for_apply_inverse*
(mne) ❯ git grep -B2 -i 'tmp' | grep fixt -A2 
mne/datasets/sleep_physionet/tests/test_physionet.py-@pytest.fixture(scope='session')
mne/datasets/sleep_physionet/tests/test_physionet.py:def physionet_tmpdir(tmpdir_factory):
mne/datasets/sleep_physionet/tests/test_physionet.py-    """Fixture exposing a temporary directory for testing."""
--
mne/io/eeglab/tests/test_eeglab.py-@pytest.fixture(scope='session')
mne/io/eeglab/tests/test_eeglab.py:def one_chanpos_fname(tmpdir_factory):
--
--
mne/tests/test_annotations.py-@pytest.fixture(scope='session')
mne/tests/test_annotations.py:def dummy_annotation_csv_file(tmpdir_factory):
--
--
mne/tests/test_annotations.py-@pytest.fixture(scope='session')
mne/tests/test_annotations.py:def dummy_broken_annotation_csv_file(tmpdir_factory):
--
--
mne/tests/test_annotations.py-@pytest.fixture(scope='session')
mne/tests/test_annotations.py:def dummy_annotation_txt_file(tmpdir_factory):
--
--
mne/tests/test_annotations.py-@pytest.fixture(scope='session')
mne/tests/test_annotations.py:def dummy_annotation_txt_header(tmpdir_factory):
--
--
mne/tests/test_annotations.py-@pytest.fixture(scope='session')
mne/tests/test_annotations.py:def dummy_annotation_txt_file_with_orig_time(tmpdir_factory):
--
```

The thing should be something like:
```py
@pytest.mark.parametrize('fname, EXPECTED_MEAS, EXPECTED_STR', [
    pytest.param(vhdr_path, [1384359243, 794231], '2013-11-13 16:14:03 GMT' )
    pytest.param(vhdr_path, [1384359243, 794231], '2013-11-13 16:14:03 GMT' )
    pytest.param(vhdr_path, [1384359243, 794231], '2013-11-13 16:14:03 GMT' )
    pytest.param(vhdr_path, [1384359243, 794231], '2013-11-13 16:14:03 GMT' )
])
def test_vmrk_valid_meas_date(fname, EXPECTED_MEAS, EXPECTED_STR, warns):
    """Test successful extraction of measurement date."""
    # Test file that does have a specific date
    raw = read_raw_brainvision(fname)
    assert_allclose(raw.info['meas_date'], EXPECTED_MEAS)
    assert EXPECTED_STRING in repr(raw.info)


@pytest.mark.parametrize('fname', [vhdr_bad_date, foo, bar, xx])
def test_vmrk_wrong_meas_date(fname, warns):
    raw = read_raw_brainvision(vhdr_bad_date)
    assert raw.info['meas_date'] is None
    assert 'unspecified' in repr(raw.info)
```

Reply:
IMHO splitting the valids and invalids simplifies it.